### PR TITLE
[terraform] 0.12 support

### PR DESCRIPTION
**Pull request by osterman**
## what
* Hacks to get `terraform-docs` to work with HCL2
* Drop `json2hcl` which won't support HCl2

## why
* Terraform 0.12 use HCL2 which is not backwards compatible 

## references
* https://github.com/segmentio/terraform-docs/issues/62
* https://github.com/kvz/json2hcl/issues/12

### CODE REVIEW

File: test/terraform/module-pinning.bats [hunk 1]
```diff
@@ -1,16 +1,22 @@
...
 
 @test "check if terraform modules are properly pinned" {
   skip_unless_terraform
-  run bash -c "cat *.tf | json2hcl -reverse | jq -r '..|select(.source?)|.source' | sort -u | sed 's/^.*?ref=//' | grep -Ev '^(tags/[0-9]+\\.[0-9]+.*|)$$'"
-  log "$output"
-  [ $status -ne 0 ]
+  grep -Eo '^\s*source\s*=\s*"(.*?)"' *.tf | cut -d'"' -f2 | sort -u > $TMPFILE
+  if [ -s $TMPFILE ]; then
+    # Verify the modules are pinned to `tags/x.y` or nothing at all (maybe using `version` parameter instead)
+    sed 's/^.*?ref=//' < $TMPFILE | grep -E '^(tags/[0-9]+\.[0-9]+.*|)$'
```
Comment (aknysh) [inline test/terraform/module-pinning.bats]: why in `'^(tags/[0-9]+\.[0-9]+.*|)$'`, the first `.` escaped `\.`, but the second one not?

File: test/terraform/validate.bats [hunk 2]
```diff
@@ -1,9 +1,25 @@
...
+  unset TF_CLI_ARGS_plan
+}
+
 @test "check if terraform code is valid" {
   skip_unless_terraform
-  run terraform validate -check-variables=false
-  log "$output"
-  [ $status -eq 0 ]
-  [ -z "$output" ]
+  if [[ "`terraform version | head -1`" =~ 0.12 ]]; then
```
Comment (aknysh) [inline test/terraform/validate.bats]: why the `.` is not escaped here in `=~ 0.12`, but escaped in https://github.com/cloudposse/test-harness/pull/7/files#diff-a301e3910214949a4fe851f846db12a0R50

Comment (osterman) [reply test/terraform/validate.bats]: Good Catch

File: test/terraform/module-pinning.bats (hunk 1 above)
Comment (osterman) [reply test/terraform/module-pinning.bats]: The second one is not matching a `.`, but saying anything else (e.g `1.0` or `1.0-alpha` or 1.0.2`, etc)

File: test/terraform/validate.bats (hunk 2 above)
Comment (osterman) [reply test/terraform/validate.bats]:
```suggestion
  if [[ "`terraform version | head -1`" =~ 0\.12 ]]; then
```

**Pull request merged**